lib/CloudinaryPlugin.js

```diff
@@ -0,0 +1,333 @@
+ * limitations under the License.
+ */
+
+const Cloudinary = require('cloudinary').v2;
+
+class CloudinaryPlugin {
+
+  constructor() {
+    this.context = null;
+  }
+
+  init(config, context) {
+    this.context = context;
+
+    this.controllers = {
+      assetsController: {
+        search: 'search',
+        rename: 'rename',
+        destroy: 'destroy',
+        transform: 'transform'
+      },
+
+      tagsController: {
+        add_tag: 'add_tag',
+        remove_tag: 'remove_tag',
+        remove_all_tags: 'remove_all_tags',
+        replace_tag: 'replace_tag'
+      },
+    };
+
+    this.routes = [
+      {
+        verb: 'get',
+        url: '/assets/:expression',
+        controller: 'assetsController',
+        action: 'search'
+      },
+      {
+        verb: 'post',
+        url: '/assets/transform',
+        controller: 'assetsController',
+        action: 'transform'
+      },
+      {
+        verb: 'put',
+        url: '/assets',
+        controller: 'assetsController',
+        action: 'rename'
+      },
+      {
+        verb: 'delete',
+        url: '/assets',
+        controller: 'assetsController',
+        action: 'destroy'
+      },
+      {
+        verb: 'post',
+        url: '/tags',
+        controller: 'tagsController',
+        action: 'add_tag'
+      },
+      {
+        verb: 'put',
+        url: '/tags',
+        controller: 'tagsController',
+        action: 'replace_tag'
+      },
+      {
+        verb: 'delete',
+        url: '/tags',
+        controller: 'tagsController',
+        action: 'remove_tag'
+      },
+      {
+        verb: 'delete',
+        url: '/tags/remove_all',
+        controller: 'tagsController',
+        action: 'remove_all_tags'
+      }
+    ];
+
+    this.config = config;
+
+    const { NODE_ENV } = process.env;
+    const { cloudName, APIKey, APISecret } = this.context.secrets.cloudinary || {};
+
+    if (!APIKey || !APISecret || !cloudName) {
+      if (NODE_ENV !== 'development') {
+        throw new this.context.errors.InternalError('Cloudinary Plugin cannot be used without the ApiKey, the ApiSecret and the CloudName of your Cloudinary Account');
+      } else {
+        this.context.log.error('Cloudinary Plugin cannot be used without the ApiKey, the ApiSecret and the CloudName of your Cloudinary Account.\nUsing this plugin outside of development without those keys will prevent Kuzzle from starting.');
+      }
+    } else {
+      Cloudinary.config({
+        cloud_name: cloudName,
+        api_key: APIKey,
+        api_secret: APISecret
+      });
+    }
+
+  }
+
+  /**
+   * Controller: Assets
+   * Action: search
+   *  
+   * Get the list of assets corresponding to the given expression.
+   * 
+   * @param {Request} request 
+   * 
+   * @see https://cloudinary.com/documentation/search_api
+   */
+  async search(request) {
+    const result = await Cloudinary.search
+      .expression(this._getArg(request.input.args, 'expression'))
```

Comment:
Shouldn't we support additional cloudinary parameters? At least, the `next_cursor` one should be in IMO.